src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -402,22 +403,34 @@ public static String escapeAllControlChars(String text) {
      * Checks Javadoc comment it's in right place.
      * From Javadoc util documentation:
      * "Placement of comments - Documentation comments are recognized only when placed
-     * immediately before class, interface, constructor, method, or field
+     * immediately before class, interface, constructor, method, field or annotation field
      * declarations -- see the class example, method example, and field example.
-     * Documentation comments placed in the body of a method are ignored. Only one
+     * Documentation comments placed in the body of a method are ignored. Only [the last] one
      * documentation comment per declaration statement is recognized by the Javadoc tool."
      *
      * @param blockComment Block comment AST
      * @return true if Javadoc is in right place
      */
     private static boolean isCorrectJavadocPosition(DetailAST blockComment) {
-        return BlockCommentPosition.isOnClass(blockComment)
-                || BlockCommentPosition.isOnInterface(blockComment)
-                || BlockCommentPosition.isOnEnum(blockComment)
-                || BlockCommentPosition.isOnMethod(blockComment)
-                || BlockCommentPosition.isOnField(blockComment)
-                || BlockCommentPosition.isOnConstructor(blockComment)
-                || BlockCommentPosition.isOnEnumConstant(blockComment)
-                || BlockCommentPosition.isOnAnnotationDef(blockComment);
+        // We must be sure that after this one there are no other documentation comments.
+        DetailAST sibling = blockComment;
+        do {
```

Comment:
does it make more sense to change this from `do {} while()` to `while () {}`?
if seems weird to see `if (sibling != null) {` and `} while (sibling != null);`.